src/core/symbology/qgsmarkersymbollayer.cpp

```diff
@@ -2683,24 +2683,47 @@ void QgsRasterMarkerSymbolLayer::renderPoint( QPointF point, QgsSymbolRenderCont
   if ( !p )
     return;
 
-  bool hasDataDefinedSize = false;
-  double scaledSize = calculateSize( context, hasDataDefinedSize );
-  double width = context.renderContext().convertToPainterUnits( scaledSize, mSizeUnit, mSizeMapUnitScale );
-  bool hasDataDefinedAspectRatio = false;
-  double aspectRatio = calculateAspectRatio( context, scaledSize, hasDataDefinedAspectRatio );
-  double height = width * ( preservedAspectRatio() ? defaultAspectRatio() : aspectRatio );
-
-  //don't render symbols with size below one or above 10,000 pixels
-  if ( static_cast< int >( width ) < 1 || 10000.0 < width )
-  {
-    return;
-  }
-
   QString path = mPath;
   if ( mDataDefinedProperties.isActive( QgsSymbolLayer::PropertyName ) )
   {
     context.setOriginalValueVariable( mPath );
     path = mDataDefinedProperties.valueAsString( QgsSymbolLayer::PropertyName, context.renderContext().expressionContext(), mPath );
+  }
+
+  if ( path.isEmpty() )
+    return;
+
+  double width = 0.0;
+  double height = 0.0;
+  double scaledSize = 0.0;
+  double aspectRatio = 0.0;
+
+  // RenderPercentage Unit Type takes original image size
+  if ( mSizeUnit == QgsUnitTypes::RenderPercentage )
+  {
+    QSize size = QgsApplication::imageCache()->originalSize( path );
+    if ( size.isNull() || !size.isValid() || size.width() == 0 || size.height() == 0 )
+      return;
+
+    scaledSize = ( mSize * static_cast<double>( size.width() ) ) / 100.0;
+    width = context.renderContext().convertToPainterUnits( scaledSize, mSizeUnit, mSizeMapUnitScale );
```

Comment:
This call has no effect when mSizeUnit is percentage, we could skip it

Reply:
Thank you for your advises!
I tried to stick with the original code style as much as possible.
Unfortunately I have no much experience in the "pulling requests" - it's my first one.
So, what should I do next?

Reply:
Do I need to make a new Pull Request, or I should to change existing one?